include/swift/AST/Identifier.h

```diff
@@ -233,6 +233,12 @@ class DeclBaseName {
     return !isSpecial() && getIdentifier().isEditorPlaceholder();
   }
 
+  /// A representation of the name to be displayed to users. May be ambiguous
+  /// between identifiers and special names
+  StringRef userFacingStr() const {
```

Comment:
I'll go with `userFacingName` since it's not only used for diagnostics but also for some other stuff like sorting names.